consumer/src/main/kotlin/streams/procedures/StreamsSinkProcedures.kt

```diff
@@ -35,6 +48,80 @@ class StreamsSinkProcedures {
         return data.map { StreamResult(mapOf("data" to it)) }.stream()
     }
 
+    @Procedure("streams.sink.start")
+    fun sinkStart(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        try {
+            streamsEventSink?.start()
+            return sinkStatus()
+        } catch (e: Exception) {
+            log?.error("Cannot start the Sink because of the following exception", e)
+            return Stream.concat(sinkStatus(),
+                    Stream.of(StreamsSinkDTO("exception", ExceptionUtils.getStackTrace(e))))
+        }
+    }
+
+    @Procedure("streams.sink.stop")
+    fun sinkStop(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        try {
+            streamsEventSink?.stop()
+            return sinkStatus()
+        } catch (e: Exception) {
+            log?.error("Cannot stopped the Sink because of the following exception", e)
+            return Stream.concat(sinkStatus(),
+                    Stream.of(StreamsSinkDTO("exception", ExceptionUtils.getStackTrace(e))))
+        }
+    }
+
+    @Procedure("streams.sink.restart")
+    fun sinkRestart(): Stream<StreamsSinkDTO> {
+        val stopped = sinkStop().collect(Collectors.toList())
+        val hasError = stopped.stream().anyMatch { it.name == "exception" }
+        if (hasError) {
+            return stopped.stream()
+        }
+        return sinkStart()
+    }
+
+    @Procedure("streams.sink.config")
+    fun sinkConfig(): Stream<StreamsSinkDTO> {
+        checkEnabled()
+        checkLeader()
+        val configMap = JSONUtils.asMap(streamsSinkConfiguration)
+                .filterKeys { it != "topics" && it != "enabled" && it != "proceduresEnabled" && !it.startsWith("check") }
+                .mapKeys { it.key.toPointCase() }
+                .mapKeys {
+                    when (it.key) {
+                        "error.config" -> "streams.sink.errors"
```

Comment:
looks as if this config listing / renaming (parse / output) should be in a central place and not hidden away in the procedure

so if something changes there we need to change it only in one place.

Reply:
moved into a method inside the `StreamsSinkConfiguration` class